src/rashdf/plan.py

```diff
@@ -27,6 +69,611 @@ def __init__(self, name: str, **kwargs):
+        self,
+        mesh_name: str,
+        var: SummaryOutputVar,
+        time_unit: str = "days",
+        round_to: str = "0.1 s",
+    ) -> np.ndarray[np.datetime64]:
+        """Return an array of times for min/max summary output data.
+
+        Parameters
+        ----------
+        mesh_name : str
+            The name of the 2D flow area mesh.
+        var : SummaryOutputVar
+            The summary output variable to retrieve.
+        time_unit : str, optional
+            The time unit for the maximum water surface elevation data.
+            Default: "days".
+        round_to : str, optional
+            The time unit to round the datetimes to. Default: "0.1 s" (seconds).
+
+        Returns
+        -------
+        np.ndarray[np.datetime64]
+            An array of times for the maximum water surface elevation data.
+        """
+        start_time = self._simulation_start_time()
+        max_ws_group = self._mesh_summary_output_group(mesh_name, var)
+        time_unit = self._summary_output_min_max_time_unit(max_ws_group)
+        max_ws_raw = max_ws_group[:]
+        max_ws_times_raw = max_ws_raw[1]
+        # we get weirdly specific datetime values if we don't round to e.g., 0.1 seconds;
+        # otherwise datetimes don't align with the actual timestep values in the plan file
+        max_ws_times = ras_timesteps_to_datetimes(
+            max_ws_times_raw, start_time, time_unit=time_unit, round_to=round_to
+        )
+        return max_ws_times
+
+    def _mesh_summary_output_min_max(
+        self,
+        var: SummaryOutputVar,
+        value_col: str = "value",
+        time_col: str = "time",
+        round_to: str = "0.1 s",
+    ) -> DataFrame:
+        """Return the min/max values and times for a summary output variable.
+
+        Valid for:
+        - Maximum Water Surface
+        - Minimum Water Surface
+        - Maximum Face Velocity
+        - Minimum Face Velocity
+        - Cell Maximum Water Surface Error
+
+        Parameters
+        ----------
+        var : SummaryOutputVar
+            The summary output variable to retrieve.
+        round_to : str, optional
+            The time unit to round the datetimes to. Default: "0.1 s" (seconds).
+
+        Returns
+        -------
+        DataFrame
+            A DataFrame with columns 'mesh_name', 'cell_id' or 'face_id', 'value', and 'time'.
+        """
+        dfs = []
+        for mesh_name, cell_count in self._2d_flow_area_names_and_counts():
+            values = self._mesh_summary_output_min_max_values(mesh_name, var)
+            times = self._mesh_summary_output_min_max_times(
+                mesh_name, var, round_to=round_to
+            )
+            if var in [
+                SummaryOutputVar.MAXIMUM_FACE_VELOCITY,
+                SummaryOutputVar.MINIMUM_FACE_VELOCITY,
+            ]:
+                geom_id_col = "face_id"
+            else:
+                # For whatever reason there seem to be more values reported in the
```

Comment:
@zherbz Great question. I've taken this at @bates-j's word without looking closely at it myself, but after looking into it today I think I understand how it works in RAS.

* The "number of cells" in a mesh comes from column `Cell Count` in the table: `/Geometry/2D Flow Areas/Attributes`
* The array `/Geometry/2D Flow Areas/Cell Points` contains cell coordinates.
* There's another array which appears to contain something similar, but with a subtle difference. `/Geometry/2D Flow Areas/<mesh-name>/Cells Center Coordinate` has all the same points as `Cell Points` above, but also includes extra points on the perimeter of the 2D flow area, appended to the array. This is visible if you extract those points and symbolize them by the index number.

![image](https://github.com/fema-ffrd/rashdf/assets/5504267/b3327547-1279-4578-af18-c0df30ba939d)
![image](https://github.com/fema-ffrd/rashdf/assets/5504267/17706ad7-146c-45e7-a0e8-02d25e9fd8d9)

* The cells along the perimeter get included in the output arrays, but the output values are nonsense. Here are the same points with Max WS symbolized. All of the Max WS values along the perimter are 0.0 (dark blue):

![image](https://github.com/fema-ffrd/rashdf/assets/5504267/e5d4fb3c-7edc-4b6b-b1f9-f8833f35a603)

Here's how it looks in the attribute table -- the break between "real" cells and additional points from `Cells Center Coordinate` that appear in the output arrays but with bogus or missing values (the `2D Flow Areas/Attributes` table lists a `Cell Count` of 1066 for this area):

![image](https://github.com/fema-ffrd/rashdf/assets/5504267/b0cfd377-4451-4c8a-b7d1-b4a534af2673)

So, there are extra values in the summary output arrays in the HDF file that need to be thrown out because they correspond to non-cells along the edge of the 2D flow area.

Reply:
@zherbz I'll update the comment in the code to point this out, instead of my current "🤷‍♂️" comment.

Reply:
@zherbz updated the comment that you noted; marking this as resolved.